src/circuits/runcircuit.jl

```diff
@@ -125,12 +125,8 @@ function runcircuit(
   # which was added using the `insertnoise` function. If so, one should call directly
   # the `choimatrix` function.
   circuit_tensors = buildcircuit(hilbert, circuit; noise, device, eltype)
-  if circuit_tensors isa Vector{<:ITensor}
-    inds_sizes = [length(inds(g)) for g in circuit_tensors]
-  else
-    inds_sizes = vcat([[length(inds(g)) for g in layer] for layer in circuit_tensors]...)
-  end
-  noiseflag = any(x -> x % 2 == 1, inds_sizes)
+  layers = circuit_tensors isa Vector{<:ITensor} ? (circuit_tensors,) : circuit_tensors
```

Comment:
Even better would be to split off into a function:
```julia
to_layers(ts::Vector{ITensor}) = [ts]
to_layers(ts::Vector{Vector{ITensor}}) = ts
```
I'm weary of using a `Tuple` since our convention is that a layered circuit is nested layers of `Vector`, though I understand it avoids an allocation. I don't think we're worried about the performance of this part of the code though.

Reply:
That makes sense. I'm guessing that allocating here would change the efficiency of computing `noise_flags` very little if at all.

Would `to_layers` be more generally useful? You could use it elsewhere to support the convention a bit. If not, you might consider `[circuit_tensors] : circuit_tensors`. 

In any case, part of the point of this PR is that code serves as an example. Even though using a `Tuple` doesn't affect the behavior here, someone could copy the code or idea to another location where your convention is expected to be followed. So using an `Array` makes sense.

Furthermore, there are plans to add more escape analysis to the Julia compiler in the not too distant future. I imagine the compiler will do `[x]` $\rightarrow$ `(x,)` for you in simple cases like this.

Reply:
I think it's better to move into a function, I could imagine that would be useful in other places (and I could imagine expanding the functionality to convert other structures into our standard layered circuit format).